### DISCUSSION TITLE
Decimals being truncated in EF Core

### DISCUSSION THREAD

**Discussion by Brian-Con** [Azure/azure-mobile-apps#669] [Q&A]
Hi,

In EF Core, decimal values are truncated to 2 decimal places. EF has the same behavior, but in EF you can override that behavior with the following code:

```c#
 public class DbContextConfiguration : DbConfiguration
    {
        public DbContextConfiguration()
        {
            var providerInstance = SqlProviderServices.Instance;
            SqlProviderServices.TruncateDecimalsToScale = false;
            this.SetProviderServices(SqlProviderServices.ProviderInvariantName, SqlProviderServices.Instance);
        }
    }
    [DbConfigurationType(typeof(DbContextConfiguration))]
```

Is there an equivalent to this in EF Core - A central place when you can configure the code not to truncate decimals?

I'm aware of the **[Precision(15,6)]** keyword which you can use to override the behavior at the individual Property level, but I'd rather not go down the path of having to set this attribute for each Property if there is a better option.

```c#
        [Precision(15,6)]
        public decimal? HZ_Qty { get; set; }
        public string? HZ_UOM { get; set; }
        [Precision(15, 6)]
        public decimal? HZ_OriginalQty { get; set; }
        public string? HZ_OriginalUOM { get; set; }
``` 

Thanks,
Brian

**Comment by adrianhall** [comment]
This isn't an "Azure Mobile Apps" issue - I can't assist with EF Core issues.  Suggest you post this on the github.com/dotnet/ef repo.

**Reply by Brian-Con** [reply]
Thanks for your guidance.
